src/network/networkd-route-nexthop.c

```diff
@@ -539,6 +544,11 @@ static int append_nexthop_one(const Route *route, const RouteNextHop *nh, struct
         assert(rta);
         assert(*rta);
 
+        if (route->family == AF_INET6 && !in_addr_is_set(nh->family, &nh->gw))
+                return log_warning_errno(
```

Comment:
Please drop. I guess it should work. The reason that currently it fails should be you do not set `RTA_GATEWAY` when no address is specified.